src/main/java/ch/njol/skript/expressions/ExprBossBarFlags.java

```diff
@@ -0,0 +1,110 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.boss.BarFlag;
+import org.bukkit.boss.BossBar;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Name("Boss Bar Flags")
+@Description("The flags of a boss bar. These flags control the behavior of the boss bar.")
+@Examples({"add darken sky to flags of {_bar}t"})
+@Since("INSERT VERSION")
+public class ExprBossBarFlags extends PropertyExpression<BossBar, BarFlag> {
+
+	static {
+		register(ExprBossBarFlags.class, BarFlag.class, "flags", "bossbars");
```

Comment:
i don't like this being so generic when things like item flags exist and i'm sure addons like to use flags too. Could be `[boss]bar flags` or could be an any-flagged syntax, though.

Reply:
We might want to wait for AnyX for this PR